Automattic/jetpack
# Jetpack SSO: enable 'Match by Email' by default

## Proposed changes:
This PR changes the default behavior of "Match by Email" functionality. This option has historically been enabled by default since the beginning of times. In #27102 we made the default be `off` for consistency with its representation in the UI, and it stayed like this for the last 8 months. But there were consequences. See p9GBOq-5Re-p2. With this update, the option will be reactivated by default for all users who have never modified this particular setting.


**Before:** Match by Email disabled by default
**After:** Match by Email is enabled by default, and only disabled if an admin manually disables it.


### Other information:


## Jetpack product discussion
p9GBOq-5Re-p2
#27102
p1690992724021369-slack-CDD9LQRSN
p9o2xV-2zY-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:
1. Create a new site and enable SSO in "Jetpack -> Settings -> Security". Do not touch the "Match by email" setting, leave it enabled.
2. Create a user with email matching another WPCOM account (let's call them "user 2").
4. Use the "user 2" account to log into WP-admin for the first time via SSO, you should succeed.
5. Go to "Jetpack -> Dashboard" and disconnect "user 2" from WPCOM.
6. Go to "Jetpack -> Settings -> Security" and disable the "Match by email" setting.
7. Try to log in as "user 2" via SSO again. You should see the "We couldn't find your account" error.
9. Go to "Jetpack -> Settings -> Security" and enable "Match by email".
10. Disconnect "user 2" from WPCOM.
11. Logout in Jetpack and WPCOM as "user 2" and try to login via SSO, you should succeed.
